drivers/ethernet/eth_native_posix.c

```diff
@@ -441,14 +441,23 @@ static struct device *eth_get_ptp_clock(struct device *dev)
 }
 #endif
 
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
+static struct net_stats_eth *get_stats(struct net_if *iface)
+{
+	struct eth_context *context = net_if_get_device(iface)->driver_data;
+
+	return &(context->stats);
+}
+#endif
+
 static const struct ethernet_api eth_if_api = {
 	.iface_api.init = eth_iface_init,
 	.iface_api.send = eth_send,
 
 	.get_capabilities = eth_posix_native_get_capabilities,
 
 #if defined(CONFIG_NET_STATISTICS_ETHERNET)
-	.stats = &eth_context_data.stats,
+	.get_stats = &get_stats,
```

Comment:
Function name is already its address and you should not use & here. Just do `.get_stats = get_stats,`

Reply:
Done.